HomeAutomationServer/Models/AppCache.cs

```diff
@@ -46,51 +51,56 @@ static public bool AddDeviceBlob_DEBUG(JObject blob)
 
 		/*************** END DEBUG MODE METHOD *************/
 
-        static public bool AddDeviceBlob(JObject blob)
-        {
-            deviceBlobs[(string)blob["deviceID"]] = blob;
-
-            if (deviceBlobs[(string)blob["deviceID"]] == blob)
-            {
-                try
-                {
-                    AmazonSimpleNotificationServiceClient snsClient = new AmazonSimpleNotificationServiceClient("AKIAJM2E3LGZHJYGFSQQ", "p3Qi8DAXj+XHAH+ny7HrlRyleBs5V5DJv77zKK3T", Amazon.RegionEndpoint.USEast1);
-                    snsClient.Publish("arn:aws:sns:us-east-1:336632281456:MyTopic", "New Device Updates");
-                }
-
-                catch (Exception ex)
-                {
-                    LogFile.AddLog("AppCache -- Could not send Push Notification: " + ex.Message + "\n");
-                    return false;
-                }
-
-                return true;
-            }
-
-            return false;
-        }
-
-        static public JToken GetDeviceBlob(string deviceID)
-        {
-            JToken blob = deviceBlobs[deviceID];
-            deviceBlobs[deviceID].Remove();
-            return blob;
-        }
-
-        static public JArray GetAllBlobs()
-        {
-            JArray blobs = deviceBlobs;
-            deviceBlobs.RemoveAll();
-            return blobs;
-        }
-
-        static public int GetBlobCount()
-        {
-            return deviceBlobs.Count;
-        }
-
-        ////////////////////////////////////////////////////////////////////////////////////////
-        //
-        // other blob methods
-    }
+		static public bool AddDeviceBlob(JObject blob)
+		{
+			string blob_string = blob.ToString ();
+			Device dev = Interfaces.DeserializeDevice(blob_string, null, null, new TimeFrame()); // to convert the JSON blob to an actual Device object.
+			string devID = (string)blob["deviceID"];
+
+			deviceBlobs.Add(devID, dev);
+
+			if (deviceBlobs.Contains(dev)) {
+				try
+				{
+					AmazonSimpleNotificationServiceClient snsClient = new AmazonSimpleNotificationServiceClient("AKIAJM2E3LGZHJYGFSQQ", "p3Qi8DAXj+XHAH+ny7HrlRyleBs5V5DJv77zKK3T", Amazon.RegionEndpoint.USEast1);
+					snsClient.Publish("arn:aws:sns:us-east-1:336632281456:MyTopic", "New Device Updates");
+				}
+
+				catch (Exception ex)
+				{
+					LogFile.AddLog("AppCache -- Could not send Push Notification: " + ex.Message + "\n");
+					return false;
+				}
+
+				return true;
+			}
+
+			return false;
+		}
+
+		static public JToken GetDeviceBlob(string deviceID)
```

Comment:
We wrote our API under the impression the app could just give us the device ID (because it's unique) in order to get a device update. From what you're saying it sounds like we'll have to adjust our API and require the app team to provide a FullID object -- or at least all three ID's in some way (so they can be parsed into a FullID object)? Do i have this right?